crates/nargo/src/cli/prove_cmd.rs

```diff
@@ -46,9 +48,16 @@ pub fn prove_with_path<P: AsRef<Path>>(
     show_ssa: bool,
     allow_warnings: bool,
 ) -> Result<Option<PathBuf>, CliError> {
-    let (compiled_program, solved_witness) =
+    let (mut compiled_program, solved_witness) =
         compile_circuit_and_witness(program_dir, show_ssa, allow_warnings)?;
 
+    // Since the public outputs are added into the public inputs list
+    // There can be duplicates. We keep the duplicates for when one is
+    // encoding the return values into the Verifier.toml
+    // However, for creating a proof, we remove these duplicates.
+    compiled_program.circuit.public_inputs =
+        dedup_public_input_indices(compiled_program.circuit.public_inputs);
```

Comment:
Perhaps this should be performed within `compile_circuit_and_witness` so there is no expectation for users calling that function to always have to fix the inputs afterward?

Reply:
I agree, this has nothing to do with `prove_with_path` so it makes sense to put it in `compile_circuit_and_witness`